### DISCUSSION TITLE
How to pass a Notification to component?

### DISCUSSION THREAD

**Discussion by ithuis** [livewire/livewire#7861] [1. Help]
Is there a way to pass along something like a notification to a livewire component

i cant use properties  because they need to be public and type "Illuminate\Notifications\Notification" is not supported for "public" properties. 

i would like to be able to set a property of any type, which doesn't need to be accessible by the frontend

``` 
<livewire:component :notification="new TestNotofication($model)" />
```

**Comment by joshhanley** [comment]
@ithuis the problem will be you are assigning the notification to a public property inside your component which Livewire doesn't support. So either don't assign it to a public property or make your notification class `wireable` or make a custom `synth` for all of your notification classes. 

Wireable: https://livewire.laravel.com/docs/properties#wireables
Synths: https://livewire.laravel.com/docs/synthesizers

**Reply by ithuis** [reply]
@joshhanley my goal is to pass  a notification and only use it in the php.  

I would prefer not to use wireables nor synthesizer because that would imply overhead in the sync to the frontend. 
And i cant set a property which is not set as public.

**Reply by joshhanley** [reply]
@ithuis yeah fair enough. Just pass it through the mount method and don't make it/ don't assign it to a public property. If you can share you component class I can show you what I mean.

**Reply by ithuis** [reply]
this is it! thank you, ik completely forgot about that..

this works.. 
```
  public function mount(Notification $notification)
    {
        dd($notification);
    }
```

**Reply by joshhanley** [reply]
@ithuis sweet! No worries 🙂

**Reply by ithuis** [reply]
@joshhanley sorry to get back to this. But when is set private properties in the mount, there value is lost.
I have setup an example for you.

https://laravelplayground.com/#/snippets/148833eb-ebe5-4491-a660-5814a063acb0

**Reply by joshhanley** [reply]
@ithuis You need to use public properties to persist state between requests. See here https://livewire.laravel.com/docs/properties

**Reply by ithuis** [reply]
ok, so what i am after is just not possible ( for now ).. thanks anyway, amazing support!

**Reply by joshhanley** [reply]
@ithuis No worries! You could make use of session or cache. I tend to use computed properties (which can use the cache) to track data that I don't want sent to the front end, but I want available in my class.
